### PR TITLE
DOCS-5203 Environment Limitations

### PR DESCRIPTION
**Pull request by paigehargrave**

### CODE REVIEW

**Comment by paigehargrave** [comment]
Suggested updates for https://www.mulesoft.org/jira/browse/DOCS-5203.

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -34,4 +35,4 @@
  ** xref:config-add-proxy-whitelist.adoc[Add Whitelist Addresses for API Platform Proxies]
  ** xref:demo-ldap-server.adoc[Configure the Demo LDAP Server]
  ** xref:ext-analytics-elk.adoc[Analyze Business and API Data using ELK]
-* xref:upgrade.adoc[Migrating Anypoint Platform Private Cloud]
\ No newline at end of file
+* xref:upgrade.adoc[Migrating Anypoint Platform Private Cloud]
```
Comment (federicoboerr) [inline modules/ROOT/nav.adoc:38]: Is this part of this PR?

File: modules/ROOT/nav.adoc [hunk 2]
```diff
@@ -9,6 +9,7 @@
  ** xref:prereq-verify-device.adoc[Verify Device Requirements]
  ** xref:prereq-network.adoc[Network Requirements]
  ** xref:prereq-software.adoc[About Software Requirements]
+ ** xref:prereq-env.adoc[Environment Limitations]
```
Comment (federicoboerr) [inline modules/ROOT/nav.adoc:12]: Should this be included here or in the release notes? I'm not sure about the pros and cons.

**Review by federicoboerr — COMMENTED** [review]
some comments

File: modules/ROOT/pages/prereq-env.adoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
+= Environment Limitations
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Adhere to the follow recommendations and limitations when setting up your environment and deploying apps:
```
Comment (hannanelson) [inline modules/ROOT/pages/prereq-env.adoc]: following recommendations

File: modules/ROOT/pages/prereq-env.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+** Production: 6 Nodes of 8 cores and 32GB 
+
+* Sizing
+
+** ARM:
+
+*** Applications per Environment 
+
+**** Less than 200
```
Comment (hannanelson) [inline modules/ROOT/pages/prereq-env.adoc]: Less > Fewer?

File: modules/ROOT/pages/prereq-env.adoc [hunk 5]
```diff
@@ -0,0 +1,39 @@
+= Environment Limitations
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Adhere to the follow recommendations and limitations when setting up your environment and deploying apps:
+
+* Environment:
+
+** Production: 6 Nodes of 8 cores and 32GB
```
Comment (hannanelson) [inline modules/ROOT/pages/prereq-env.adoc]: Should nodes be lowercase?

File: modules/ROOT/pages/prereq-env.adoc [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+* Sizing
+
+** ARM:
+
+*** Applications per Environment 
+
+**** Less than 200 
+
+***** Maximum: 500 Applications in a single environment
```
Comment (hannanelson) [inline modules/ROOT/pages/prereq-env.adoc]: Should Applications be lowercase (and maybe "apps"?) 

Global: look at capitalization and apply Kim's rules  :)

File: modules/ROOT/pages/prereq-env.adoc [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+** ARM:
+
+*** Applications per Environment 
+
+**** Less than 200 
+
+***** Maximum: 500 Applications in a single environment 
+**** Number of Clusters/Runtimes
+
+***** Total: 800 runtimes 2 nodes each with 400 Clusters
```
Comment (hannanelson) [inline modules/ROOT/pages/prereq-env.adoc]: Kim has been changing my numerals to words: 2 > two.

File: modules/ROOT/pages/prereq-env.adoc [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+***** Maximum: 500 Applications in a single environment 
+**** Number of Clusters/Runtimes
+
+***** Total: 800 runtimes 2 nodes each with 400 Clusters 
+**** Applications per Mule Runtime or Cluster
+
+***** Production: Maximum of 50 apps 
+***** Development: Maximum of 100 apps
+***** Recommended: Less than 25 apps
```
Comment (hannanelson) [inline modules/ROOT/pages/prereq-env.adoc]: Fewer

File: modules/ROOT/pages/prereq-env.adoc [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+*** Applications per Environment 
+
+**** Less than 200 
+
+***** Maximum: 500 Applications in a single environment 
+**** Number of Clusters/Runtimes
+
+***** Total: 800 runtimes 2 nodes each with 400 Clusters 
+**** Applications per Mule Runtime or Cluster
```
Comment (hannanelson) [inline modules/ROOT/pages/prereq-env.adoc]: Kim says "Mule" or "Mule runtime engine"

File: modules/ROOT/pages/prereq-env.adoc [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+** API Platform
+
+*** APIs under management
+
+**** Limit each management plane 2.0.0 cluster to 20000 (6 nodes, 16 cores) APIs under management and 10000 (6 nodes 8 cores), 
+in which:
+
+***** A single API implementation with one policy counts as 0.3.
+***** A single clustered proxy with up to 5 policies counts as 1.
+***** Every runtime is polling every 60 seconds.
```
Comment (hannanelson) [inline modules/ROOT/pages/prereq-env.adoc]: All runtime (engines?) poll every 60 seconds?

**Review by hannanelson — APPROVED** [review]
Some teeny edits.

File: modules/ROOT/nav.adoc (hunk 2 above)
Comment (paigehargrave) [reply modules/ROOT/nav.adoc:12]: I think it should be included here, unless it's something that's going to be fixed. Putting this in the release notes would mean classifying it as a 'Known Issue', which implies that it should be fixed. This information looks like a long term situation, which means it should go in the docs. Thoughts?

File: modules/ROOT/pages/prereq-env.adoc (hunk 3 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Got it - thank you!

File: modules/ROOT/pages/prereq-env.adoc (hunk 4 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Changed - thank you!

File: modules/ROOT/pages/prereq-env.adoc (hunk 5 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Yup - fixed.

File: modules/ROOT/pages/prereq-env.adoc (hunk 6 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Got it - thanks!

File: modules/ROOT/pages/prereq-env.adoc (hunk 7 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: OK - thanks! Got it.

File: modules/ROOT/pages/prereq-env.adoc (hunk 8 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Got it - thanks!

File: modules/ROOT/pages/prereq-env.adoc (hunk 9 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: OK - thanks! Got it.

File: modules/ROOT/pages/prereq-env.adoc (hunk 10 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Yup - clarified this. Thanks!

File: modules/ROOT/nav.adoc (hunk 1 above)
Comment (paigehargrave) [reply modules/ROOT/nav.adoc:38]: This was there before - not sure why it got flagged as a change.

**Review by federicoboerr — APPROVED** [review]
LGTM

File: modules/ROOT/pages/prereq-env.adoc [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+Adhere to the following recommendations and limitations when setting up your environment and deploying apps:
+
+* Environment:
+
+** Production: 6 nodes of 8 cores and 32GB 
+
+* Sizing
+
+** ARM:
```
Comment (federicoboerr) [inline modules/ROOT/pages/prereq-env.adoc]: Runtime Manager

File: modules/ROOT/pages/prereq-env.adoc [hunk 12]
```diff
@@ -0,0 +1,42 @@
+= Environment Limitations
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Adhere to the following recommendations and limitations when setting up your environment and deploying apps:
+
+* Environment:
+
+** Production: 6 nodes of 8 cores and 32GB
```
Comment (federicoboerr) [inline modules/ROOT/pages/prereq-env.adoc]: and 32GB `RAM`

File: modules/ROOT/pages/prereq-env.adoc [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Adhere to the following recommendations and limitations when setting up your environment and deploying apps:
+
+* Environment:
+
+** Production: 6 nodes of 8 cores and 32GB 
+** Maximum number of servers: 500
```
Comment (federicoboerr) [inline modules/ROOT/pages/prereq-env.adoc]: All these belong below under `Runtime Management`

File: modules/ROOT/pages/prereq-env.adoc [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+** Maximum number of servers per server group: 20
+** Maximum servers per cluster: 8
+
+* Sizing
+
+** Runtime Manager:
```
Comment (federicoboerr) [inline modules/ROOT/pages/prereq-env.adoc]: change to `Runtime Management`

File: modules/ROOT/pages/prereq-env.adoc [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+
+*** Applications per Environment
```
Comment (federicoboerr) [inline modules/ROOT/pages/prereq-env.adoc]: I'd say, merging with the text below as well:
```
*** Mule applications under management:
**** Recommended: around 200
**** Maximum: 500

*** Mule runtimes under management:
**** Maximum: 800 single Mule runtimes (or 400 Mule runtime clusters, consisting of 2 Mule runtimes per cluster)

*** Mule applications per Mule runtime:
***** Production: Maximum of 50 apps 
***** Development: Maximum of 100 apps
***** Recommended: Fewer than 25 apps
```

File: modules/ROOT/pages/prereq-env.adoc [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+
+***** Total: 800 runtimes, two nodes each with 400 Clusters 
+**** Applications per Mule or Cluster
+
+***** Production: Maximum of 50 apps 
+***** Development: Maximum of 100 apps
+***** Recommended: Fewer than 25 apps
+** API Platform
+
+*** APIs under management
```
Comment (federicoboerr) [inline modules/ROOT/pages/prereq-env.adoc:38]: Add: 
```
** API Management
*** APIs under management
```

**Review by federicoboerr — CHANGES_REQUESTED** [review]
some re-ordering

File: modules/ROOT/pages/prereq-env.adoc (hunk 12 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Got it - thank you!

File: modules/ROOT/pages/prereq-env.adoc (hunk 13 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Moved - thank you!

File: modules/ROOT/pages/prereq-env.adoc (hunk 14 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Got it - thanks!

File: modules/ROOT/pages/prereq-env.adoc (hunk 16 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc:38]: Got it - thanks!

File: modules/ROOT/pages/prereq-env.adoc (hunk 15 above)
Comment (paigehargrave) [reply modules/ROOT/pages/prereq-env.adoc]: Made updates.

**Review by federicoboerr — APPROVED** [review]

**Comment by paigehargrave** [comment]
@federicoboerr Thank you!

**Pull request merged**